### PR TITLE
TASK: persist ui state to local storage

### PR DESCRIPTION
**Pull request by dimaip**
Saves the `ui` part of the state to local storage on every action that
matches the `persistentActionsPattern`.

TODO:

- [x] Init other parts of the state, not only left and right sidebars

Closes: #13

### CODE REVIEW

**Comment by dimaip** [comment]
Merge after #481 and #482 or rebase

File: packages/neos-ui-redux-store/src/UI/EditModePanel/index.js [hunk 1]
```diff
@@ -30,11 +30,9 @@ export const actions = {
 // Export the reducer
 //
 export const reducer = handleActions({
-    [system.INIT]: () => $set(
+    [system.INIT]: payload => $set(
         'ui.editModePanel',
-        new Map({
-            isHidden: true
```
Comment (dimaip) [inline packages/neos-ui-redux-store/src/UI/EditModePanel/index.js]: btw, can we just throw away fallback state from here and rely on default state defined in Settings? https://github.com/neos/neos-ui/blob/master/Configuration/Settings.yaml#L87

Comment (skurfuerst) [inline packages/neos-ui-redux-store/src/UI/EditModePanel/index.js]: we could; although I quite like that you see the state structure in the reducers.

Comment (dimaip) [inline packages/neos-ui-redux-store/src/UI/EditModePanel/index.js]: I think we should do either or. It's really hard to trace the source of a particular part of state now, good source for bugs.

File: packages/neos-ui/src/localStorageMiddleware.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+//
+// Local storage middleware.
+//
+// Saves "ui" part of the state to local storage on every action
+// that matches the "persistentActionsPattern"
+//
+const localStorageMiddleware = ({getState}) => {
+    let timer = null;
+    const debounceLocalStorageTimeout = 1000;
+    const persistentActionsPatterns = ['@neos/neos-ui/UI', '@neos/neos-ui/User/Settings'];
```
Comment (dimaip) [inline packages/neos-ui/src/localStorageMiddleware.js]: This feels a bit wacky to me now, I think we need a more expressive convention of how we decide what to store in the local storage and what not.

**Pull request merged**